GPU/GLES/FragmentShaderGenerator.cpp

```diff
@@ -117,39 +117,39 @@ bool GenerateFragmentShader(const ShaderID &id, char *buffer) {
 		WRITE(p, "precision lowp float;\n");
 	} else {
 		// TODO: Handle this in VersionGEThan?
-#if !defined(FORCE_OPENGL_2_0)
-		if (gl_extensions.VersionGEThan(3, 3, 0)) {
-			fragColor0 = "fragColor0";
-			texture = "texture";
-			glslES30 = true;
-			bitwiseOps = true;
-			texelFetch = "texelFetch";
-			WRITE(p, "#version 330\n");
-		} else if (gl_extensions.VersionGEThan(3, 0, 0)) {
-			fragColor0 = "fragColor0";
-			bitwiseOps = true;
-			texelFetch = "texelFetch";
-			WRITE(p, "#version 130\n");
-			if (gl_extensions.EXT_gpu_shader4) {
-				WRITE(p, "#extension GL_EXT_gpu_shader4 : enable\n");
-			}
-		} else {
-			WRITE(p, "#version 110\n");
-			if (gl_extensions.EXT_gpu_shader4) {
-				WRITE(p, "#extension GL_EXT_gpu_shader4 : enable\n");
+		if (!forceOpenGL2_0 || gl_extensions.IsCoreContext) {
+			if (gl_extensions.VersionGEThan(3, 3, 0)) {
+				fragColor0 = "fragColor0";
+				texture = "texture";
+				glslES30 = true;
 				bitwiseOps = true;
-				texelFetch = "texelFetch2D";
+				texelFetch = "texelFetch";
+				WRITE(p, "#version 330\n");
+			} else if (gl_extensions.VersionGEThan(3, 0, 0)) {
+				fragColor0 = "fragColor0";
+				bitwiseOps = true;
+				texelFetch = "texelFetch";
+				WRITE(p, "#version 130\n");
+				if (gl_extensions.EXT_gpu_shader4) {
+					WRITE(p, "#extension GL_EXT_gpu_shader4 : enable\n");
+				}
+			} else {
+				WRITE(p, "#version 110\n");
+				if (gl_extensions.EXT_gpu_shader4) {
+					WRITE(p, "#extension GL_EXT_gpu_shader4 : enable\n");
+					bitwiseOps = true;
+					texelFetch = "texelFetch2D";
+				}
 			}
 		}
-#endif
 
 		// We remove these everywhere - GL4, GL3, Mac-forced-GL2, etc.
 		WRITE(p, "#define lowp\n");
 		WRITE(p, "#define mediump\n");
 		WRITE(p, "#define highp\n");
 	}
 
-	if (glslES30) {
+	if (glslES30 || gl_extensions.IsCoreContext) {
```

Comment:
Well, I am trying to stay away from unexpectedly breaking working compatibility profiles, but I can change this.

A core context will always be 3.x+, so it's not totally crazy.

-[Unknown]